R/rescale_weights.R

```diff
@@ -104,9 +169,63 @@ rescale_weights <- function(data, by, probability_weights, nest = FALSE) {
     data_tmp <- data
   }
 
+  fun_args <- list(
+    nest = nest,
+    probability_weights = probability_weights,
+    data_tmp = data_tmp,
+    data = data,
+    by = by,
+    weight_non_na = weight_non_na
+  )
+
+  switch(method,
+    carle = do.call(.rescale_weights_carle, fun_args),
+    do.call(.rescale_weights_kish, fun_args)
+  )
+}
+
+
+# rescale weights, method Kish ----------------------------
+
+.rescale_weights_kish <- function(nest, probability_weights, data_tmp, data, by, weight_non_na) {
+  # check argument
+  if (!is.null(by)) {
+    insight::format_warning("The `by` argument is not used for `method = \"kish\" and will be ignored.")
```

Comment:
I actually think we can use the by argument error and just apply the transformation separately to each group.